tools/cell_census_builder/src/cell_census_builder/release_gc.py

```diff
@@ -0,0 +1,146 @@
+import argparse
+import logging
+import sys
+from datetime import datetime, timedelta
+from typing import List
+
+import s3fs
+
+from .release_manifest import (
+    CensusDirectory,
+    CensusVersionName,
+    commit_release_manifest,
+    get_release_manifest,
+    validate_release_manifest,
+)
+from .util import urlcat
+
+
+def remove_releases_older_than(days: int, census_base_url: str, dryrun: bool, s3_anon: bool = False) -> None:
+    """
+    Remove old releases, commiting the change to release.json.
+
+    Current rules - delete releases where:
+    * Tag is a date older than `days` in age
+    * Tag is aliased by another tag (e.g., 'latest')
+    * There will remain at least one build (keeps _at least_ the newest)
+
+    In other words, it will only delete those releases that are not tagged (e.g., 'latest') and which
+    are older than 32 days. And it will never delete all releases.
+
+    Age of release is determined by the release version name, i.e., YYYY-MM-DD. The S3 object date or
+    other information is not utilized.
+    """
+
+    _logit(f"Delete releases older than {days} days old.", dryrun)
+
+    # Load the release manifest
+    release_manifest = get_release_manifest(census_base_url=census_base_url, s3_anon=s3_anon)
+    # validate that the manifest is safe & sane
+    validate_release_manifest(census_base_url, release_manifest, live_corpus_check=True, s3_anon=s3_anon)
+    # select build tags that can be deleted
+    rls_tags_to_delete = _find_removal_candidates(release_manifest, days)
+
+    _logit(f"Found {len(rls_tags_to_delete)} releases, older than {days} days, and not otherwise tagged.", dryrun)
+
+    # Exit if no work to do.
+    if len(rls_tags_to_delete) > 0:
+        # IMPORTANT: commit the changes to release.json before doing the delete, in case this
+        # fails for some reason (at which point, the deletions should NOT occur).
+        _update_release_manifest(release_manifest, rls_tags_to_delete, census_base_url, dryrun)
+
+        # Now delete the builds.
+        for rls_tag in rls_tags_to_delete:
+            rls_info = release_manifest[rls_tag]
+            assert isinstance(rls_info, dict)
+            uri = urlcat(rls_info["soma"]["uri"], "..")
+            _perform_recursive_delete(rls_tag, uri, dryrun)
+
+
+def _logit(msg: str, dryrun: bool) -> None:
+    logging.info(f"{'(dryrun) ' if dryrun else ''}{msg}")
+
+
+def _update_release_manifest(
+    release_manifest: CensusDirectory,
+    rls_tags_to_delete: list[CensusVersionName],
+    census_base_url: str,
+    dryrun: bool,
+) -> None:
+    new_manifest: CensusDirectory = {k: v for k, v in release_manifest.items() if k not in rls_tags_to_delete}
```

Comment:
I don't think it's a problem in practice, but could make this a deep copy, and move to a function in release_manifest.py